### PR TITLE
feat: added a sync alternative to memoize.

### PR DESCRIPTION
**Pull request by krackerman**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,3 +34,13 @@ The cached value is returned when it becomes available (important in the case wh
...
 
 Useful when multiple async operation need to access the same async information.
+
+### memoizeSync(key: String, valueFn: function)
+
+Similar to "memoize", when key is not present (or has expired), `valueFn` is called and placed into the cache.
+
+Cached value is returned when it becomes available.
+
+`valueFn` only supports Synchronous functions.
```
Comment (simlu) [inline README.md]: Not entirely true, however the below note is the important case here for async functions

File: README.md [hunk 2]
```diff
@@ -34,3 +34,13 @@ The cached value is returned when it becomes available (important in the case wh
...
+
+The cache is NOT left empty if an error is thrown at any point in `valueFn`.
```
Comment (simlu) [inline README.md]: You got this almost right. But it's still wrong :)

If you're not clear about functionality, feel free to add tests!

Comment (krackerman) [reply README.md]: I found Synchronous functions empty the cache at that key. When using Asynchronous I couldn't quite get an expect statement to work but the there seemed to be an object after an error was thrown

File: README.md [hunk 3]
```diff
@@ -34,3 +34,13 @@ The cached value is returned when it becomes available (important in the case wh
 The cache is left empty if an error is thrown at any point in `valueFn`.
 
 Useful when multiple async operation need to access the same async information.
+
+### memoizeSync(key: String, valueFn: function)
+
+Similar to "memoize", when key is not present (or has expired), `valueFn` is called and placed into the cache.
+
+Cached value is returned when it becomes available.
```
Comment (simlu) [inline README.md]:
```suggestion
Cached value is then returned.
```

**Pull request merged**